File: 이재훈/프로그래머스/Lv2/프로세스.py
from collections import deque

def solution(priorities, location):
    answer = 0
    
    cnt = 0
    priorities = deque([(i,p) for i,p in enumerate(priorities)])
    while priorities:
        
        p1 = priorities.popleft()
        is_high = True
        for p2 in priorities:
            if p2[1] > p1[1]:
                priorities.append(p1)
                is_high = False
                break
        if is_high:
            cnt+=1
            if p1[0]==location:
                return cnt
        
    return answer

File: 프로그래머스/2/42587. 프로세스/프로세스.java
import java.util.*;

class Solution {
    public int solution(int[] priorities, int location) {
        Queue<Integer> priority = new LinkedList<>();
        Queue<Integer> index = new LinkedList<>();
        
        for(int i=0; i<priorities.length; i++) {
            priority.add(priorities[i]);
            index.add(i);
        }
        int cnt = 1;
        while(!priority.isEmpty()) {
            int max = Collections.max(priority);
            int cur = priority.poll();
            int curIdx = index.poll();
            
            if(cur == max) {
                if(curIdx == location) {
                    break;
                }
                cnt++;
            } else {
                    priority.add(cur);
                    index.add(curIdx);
            }
        }
        
        return cnt;
    }
}